Makefile

```diff
@@ -79,14 +107,37 @@ bin/licensei:
 	@mkdir -p bin
 	curl -sfL https://raw.githubusercontent.com/goph/licensei/master/install.sh | bash -s -- v${LICENSEI_VERSION}
 
+bin/cosign:
+	@mkdir -p bin
+	@OS=$$(uname -s); \
+	case $$OS in \
+		"Linux") \
+			curl -sSfL https://github.com/sigstore/cosign/releases/download/v${COSIGN_VERSION}/cosign-linux-amd64 -o bin/cosign; \
+			;; \
+		"Darwin") \
+			curl -sSfL https://github.com/sigstore/cosign/releases/download/v${COSIGN_VERSION}/cosign-darwin-arm64 -o bin/cosign; \
+			;; \
+		*) \
+			echo "Unsupported OS: $$OS"; \
+			exit 1; \
+			;; \
+	esac
+	@chmod +x bin/cosign
+
 bin/goreleaser:
 	@mkdir -p bin
 	@mkdir -p tmpgoreleaser
-	curl -sfL https://goreleaser.com/static/run | VERSION=v${GORELEASER_VERSION} TMPDIR=${PWD}/tmpgoreleaser bash -s -- --version
-	mv tmpgoreleaser/goreleaser bin/
+	curl -sfL https://goreleaser.com/static/run > tmpgoreleaser/goreleaser
+	@sed -i '' -e 's|"\$$TMP_DIR/goreleaser" "\$$@"|mv "\$$TMP_DIR/goreleaser" "bin/"\nrm -rf "\$$TMP_DIR"|' tmpgoreleaser/goreleaser
+	bash tmpgoreleaser/goreleaser DISTRIBUTION=${GORELEASER_DISTRIBUTION} VERSION=v${GORELEASER_VERSION}
 	@rm -rf tmpgoreleaser
```

Comment:
this is executing the goreleaser which is not needed, only the binary needs to be pulled. The code below work
```bash
bin/goreleaser:
	@mkdir -p bin
	curl -sfL https://goreleaser.com/static/run -o bin/goreleaser
	@chmod +x bin/goreleaser
```